weblate/addons/models.py

```diff
@@ -136,7 +165,12 @@ def delete(self, using=None, keep_parents=False):
         self.store_change(Change.ACTION_ADDON_REMOVE)
         # Delete any addon alerts
         if self.addon.alert:
-            self.component.delete_alert(self.addon.alert)
+            if self.component:
+                self.component.delete_alert(self.addon.alert)
+            elif self.project:
+                for component in self.project.component_set.all():
+                    component.delete_alert(self.addon.alert)
```

Comment:
This is costly as this triggers loading all alerts for all components and there are projects with thousands of components. Delete them in the database instead:

```suggestion
                Alert.objects.filter(component__in=self.project.component_set.all(), name=self.addon.alert).delete()
```